local/node.go

```diff
@@ -38,27 +38,50 @@ type getConnFunc func(context.Context, node.Node) (net.Conn, error)
 // AvalancheGo binaries in tests
 type NodeProcess interface {
 	// Start this process
-	Start() error
+	Start(string, chan string) error
 	// Send a SIGTERM to this process
 	Stop() error
 	// Returns when the process finishes exiting
 	Wait() error
+	// Returns if the process is executing
+	Alive() bool
 }
 
 type nodeProcessImpl struct {
-	cmd *exec.Cmd
+	cmd              *exec.Cmd
+	running          bool
+	waitReturnCh     chan error
+	unexpectedStopCh chan string
 }
 
-func (p *nodeProcessImpl) Start() error {
-	return p.cmd.Start()
+func (p *nodeProcessImpl) Start(name string, unexpectedStopCh chan string) error {
+	p.unexpectedStopCh = unexpectedStopCh
```

Comment:
1. I think `node` should be part of the `nodeProcessImpl` state rather than an argument -- we already have the name when we create this struct in  `NewNodeProcess`
2.  I think the user should pass in a `chan int` rather than `chan string`, and we should send the exit code (`p.cmd.ProcessState.ExitCode()`) rather than the node name. 
3. I think we should change the naming from `unexpectedStopCh` to just `onStopChan`. The user can decide for themselves whether the stop was expected or not. If the user gets something on this channel, and they didn't call `Stop()`, they know it's unexpected.

Reply:
I also think we should guard this with a `sync.Once` to ensure it only runs once

Reply:
- regarding the Once, I added a state to the nodeProcessImpl so all functions known when they shall be called and how many times